aten/src/ATen/native/cuda/CUDALoops.cuh

```diff
@@ -151,7 +151,7 @@ __device__ inline void elementwise_kernel_helper(func_t f, policy_t policy) {
   #pragma unroll
   for (int i = 0; i < thread_work_size; i++) {
     if (policy.check_inbounds(i)) {
-      results[i] = c10::guts::apply(f, args[i]);
+      policy.store_result(i, c10::guts::apply(f, args[i]), results);
```

Comment:
Why are you doing memory access here? There is a `policy.store(results, idx);` in the end.

Reply:
Here, would `return_t` be a tuple, and `results[i] = c10::guts::apply(f, args[i]);` would just correctly store the tuple on registers? Why it needs change?

Reply:
Just because it doesn't work if `return_t` is `tuple`.

Ref: https://godbolt.org/z/uvcy8G